pkg/tagger/global.go

```diff
@@ -73,13 +105,25 @@ func Tag(entity string, cardinality collectors.TagCardinality) ([]string, error)
 // cardinality (with tags about individual containers), or at orchestrator
 // cardinality (pod/task level).
 func TagBuilder(entity string, cardinality collectors.TagCardinality, tb *util.TagsBuilder) error {
+	//TODO: defer unlock once performance overhead of defer is negligible
```

Comment:
If performance is this critical, an `atomic.Value` with a func returning a tagger instance or nil might be even faster.